ccds/hook_utils/dependencies.py

```diff
@@ -30,6 +30,43 @@
 ]
 
 
+def resolve_compatible_python_version(python_version):
+    """Resolves a Python version string to one that behaves as expected with the "compatible release"
+    operator.
+
+    Examples:
+    - 3.13 should result in ~=3.13.0 which installs the latest version of Python that is compatible
+      with 3.13, so the highest patch number for 3.13 (3.13.3 at the time of writing)
+    - 3.13.2 should result in ~=3.13.2 which ensures Python 3.13.2 is used
```

Comment:
>     - 3.13.2 should result in ~=3.13.2 which ensures Python 3.13.2 is used

This is not correct. `~=3.13.2` is equivalent to `>= 3.13.2, == 3.13.*`, so `3.13.3`, `3.13.4`, etc. will also satisfy it. 

https://packaging.python.org/en/latest/specifications/version-specifiers/#compatible-release

In the case of 3 parts, I think we want to use an exact version match.

Reply:
Aha thanks, I had misunderstood that. Updated, and confirmed that it works when providing <major>.<minor> (use `~=`) and <major>.<minor>.<patch> (use `==`).